registry_no_code_module.go

```diff
@@ -0,0 +1,284 @@
+}
+
+// NoCodeVariableOption represents a registry no-code module variable and its
+// options.
+type NoCodeVariableOption struct {
+	// Type is a public field utilized by JSON:API to
+	// set the resource type via the field tag.
+	Type string `jsonapi:"primary,variable-options"`
+
+	// Required: The variable name
+	VariableName string `jsonapi:"attr,variable-name"`
+
+	// Required: The variable type
+	VariableType string `jsonapi:"attr,variable-type"`
+
+	// Optional: The options for the variable
+	Options []string `jsonapi:"attr,options"`
+}
+
+// RegistryNoCodeModuleCreateOptions is used when creating a registry no-code module
+type RegistryNoCodeModuleCreateOptions struct {
+	// Type is a public field utilized by JSON:API to
+	// set the resource type via the field tag.
+	// It is not a user-defined value and does not need to be set.
+	// https://jsonapi.org/format/#crud-creating
+	Type string `jsonapi:"primary,no-code-modules"`
+
+	// Required: whether the no-code module should follow the latest registry module version
+	FollowLatestVersion *bool `jsonapi:"attr,follow-latest-version"`
+
+	// Required whether no-code is enabled for the registry module
+	Enabled *bool `jsonapi:"attr,enabled"`
+
+	// Required: the registry module ID
+	RegistryModule *RegistryModule `jsonapi:"relation,registry-module"`
+
+	// Optional: the registry module version pin for the no-code module
+	VersionPin *string `jsonapi:"attr,version-pin,omitempty"`
+
+	// Optional: the variable options for the no-code module
+	VariableOptions []*NoCodeVariableOption `jsonapi:"relation,variable-options,omitempty"`
+}
+
+// RegistryNoCodeModuleIncludeOpt represents the available options for include query params.
+type RegistryNoCodeModuleIncludeOpt string
+
+var (
+	// RegistryNoCodeIncludeVariableOptions is used to include variable options in the response
+	RegistryNoCodeIncludeVariableOptions RegistryNoCodeModuleIncludeOpt = "variable-options"
+)
+
+// RegistryNoCodeModuleReadOptions is used when reading a registry no-code module
+type RegistryNoCodeModuleReadOptions struct {
+	// Type is a public field utilized by JSON:API to
+	// set the resource type via the field tag.
+	// It is not a user-defined value and does not need to be set.
+	// https://jsonapi.org/format/#crud-updating
+	Type string `jsonapi:"primary,no-code-modules"`
+
+	// Optional: Include is used to specify the related resources to include in the response.
+	Include []RegistryNoCodeModuleIncludeOpt `url:"include,omitempty"`
+}
+
+// RegistryNoCodeModuleUpdateOptions is used when updating a registry no-code module
+type RegistryNoCodeModuleUpdateOptions struct {
+	// Type is a public field utilized by JSON:API to
+	// set the resource type via the field tag.
+	// It is not a user-defined value and does not need to be set.
+	// https://jsonapi.org/format/#crud-updating
+	Type string `jsonapi:"primary,no-code-modules"`
+
+	// Required: the registry module ID
```

Comment:
remove ID since the whole registry module object is passed:
```suggestion
	// Required: the registry module to use for the no-code module
```